azurerm/internal/services/apimanagement/tests/api_management_policy_resource_test.go

```diff
@@ -0,0 +1,254 @@
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementPolicyExists(data.ResourceName),
+				),
+			},
+			{
+				ResourceName:            data.ResourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"xml_link"},
+			},
+		},
+	})
+}
+
+func testCheckAzureRMApiManagementPolicyExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := acceptance.AzureProvider.Meta().(*clients.Client).ApiManagement.PolicyClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		serviceID := rs.Primary.Attributes["api_management_id"]
+		id, err := parse.ApiManagementID(serviceID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := conn.Get(ctx, id.ResourceGroup, id.ServiceName, apimanagement.PolicyExportFormatXML)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Policy (API Management Service %q / Resource Group %q) does not exist", id.ServiceName, id.ResourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on apimanagement.PolicyClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMApiManagementPolicyDestroy(s *terraform.State) error {
+	conn := acceptance.AzureProvider.Meta().(*clients.Client).ApiManagement.PolicyClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_api_management_policy" {
+			continue
+		}
+
+		serviceID := rs.Primary.Attributes["api_management_id"]
+		id, err := parse.ApiManagementID(serviceID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := conn.Get(ctx, id.ResourceGroup, id.ServiceName, apimanagement.PolicyExportFormatXML)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+
+			return err
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMApiManagementPolicy_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_api_management" "test" {
+  name                = "acctestAM-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  publisher_name      = "pub1"
+  publisher_email     = "pub1@email.com"
+  sku_name            = "Developer_1"
+}
+
+resource "azurerm_api_management_policy" "test" {
+  api_management_id = azurerm_api_management.test.id
+  xml_link          = "https://gist.githubusercontent.com/riordanp/ca22f8113afae0eb38cc12d718fd048d/raw/d6ac89a2f35a6881a7729f8cb4883179dc88eea1/example.xml"
```

Comment:
I've committed this file to master, can we download it from the repo to avoid relying on an external gist?

```suggestion
  xml_link          = "https://raw.githubusercontent.com/terraform-providers/terraform-provider-azurerm/master/azurerm/internal/services/apimanagement/tests/testdata/api_management_policy_test.xml"
```